api/src/main/java/org/ehcache/exceptions/CachePassThroughException.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.exceptions;
+
+/**
+ * A generic wrapper runtime exception that will not be caught and
+ * handled at the store level. This wrapper can be used to leak any
+ * runtime exceptions that you don't want to be caught at the store level.
+ * 
+ * @author Albin Suresh
+ */
+public class CachePassThroughException extends RuntimeException {
```

Comment:
I do not think we want that type in the API module, it is more of an internal implementation concern. Can it live in `impl` (ideal) or `core`?